File: DevExpress-Demo/CS/ChartsDemo.Wpf/Modules/ChartElements/ConstantLinesAndStripsViewModel.cs
using System.ComponentModel;

namespace ChartsDemo {
    public class ConstantLinesAndStripsViewModel : INotifyPropertyChanged {
        OperatingSurfaceTemperatureData data = new OperatingSurfaceTemperatureData();
        double optimalTemperature;

        public OperatingSurfaceTemperatureData Data { get { return data; } }
        public double OptimalTemperature {
            get { return optimalTemperature; }
            set {
                if (value < data.MinOptimalTemperature)
                    optimalTemperature = data.MinOptimalTemperature;
                else if (value > data.MaxOptimalTemperature)
                    optimalTemperature = data.MaxOptimalTemperature;
                else
                    optimalTemperature = value;
                OnPropertyChanged("OptimalTemperature");
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;

        public ConstantLinesAndStripsViewModel() {
            optimalTemperature = data.OptimalTemperature;
        }

        void OnPropertyChanged(string propertyName) {
            if (PropertyChanged != null) {
                PropertyChangedEventArgs e = new PropertyChangedEventArgs(propertyName);
                PropertyChanged(this, e);
            }
        }
    }
}

File: DevExpress-Demo/VB/ChartsDemo.Wpf/Modules/ChartElements/ConstantLinesAndStripsViewModel.vb
Imports System.ComponentModel

Namespace ChartsDemo

    Public Class ConstantLinesAndStripsViewModel
        Implements INotifyPropertyChanged

        Private dataField As OperatingSurfaceTemperatureData = New OperatingSurfaceTemperatureData()

        Private optimalTemperatureField As Double

        Public ReadOnly Property Data As OperatingSurfaceTemperatureData
            Get
                Return dataField
            End Get
        End Property

        Public Property OptimalTemperature As Double
            Get
                Return optimalTemperatureField
            End Get

            Set(ByVal value As Double)
                If value < dataField.MinOptimalTemperature Then
                    optimalTemperatureField = dataField.MinOptimalTemperature
                ElseIf value > dataField.MaxOptimalTemperature Then
                    optimalTemperatureField = dataField.MaxOptimalTemperature
                Else
                    optimalTemperatureField = value
                End If

                OnPropertyChanged("OptimalTemperature")
            End Set
        End Property

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Public Sub New()
            optimalTemperatureField = dataField.OptimalTemperature
        End Sub

        Private Sub OnPropertyChanged(ByVal propertyName As String)
            If PropertyChangedEvent IsNot Nothing Then
                Dim e As PropertyChangedEventArgs = New PropertyChangedEventArgs(propertyName)
                RaiseEvent PropertyChanged(Me, e)
            End If
        End Sub
    End Class
End Namespace
